integration_tests/src/lib.rs

```diff
@@ -1764,9 +1852,172 @@ mod test {
         Ok(())
     }
 
-    // The following tests work at the "guest" layer.  The volume
+    #[tokio::test(flavor = "multi_thread", worker_threads = 8)]
+    async fn integration_test_snapshot_backed_vol() -> Result<()> {
+        // Test using a "snapshot" (for this test, downstairs booted read-only)
+        // as a read-only parent.
+
+        const BLOCK_SIZE: usize = 512;
+
+        // boot three downstairs, write some data to them, then change to
+        // read-only.
+        let mut test_downstairs_set =
+            TestDownstairsSet::new(54082, 54083, 54084, false)?;
+
+        let mut volume = Volume::new(BLOCK_SIZE as u64);
+        volume.add_subvolume_create_guest(
+            test_downstairs_set.opts(),
+            0,
+            None,
+        )?;
+
+        volume.activate(0)?;
+
+        let random_buffer = {
+            let mut random_buffer = vec![0u8; volume.total_size()? as usize];
+            rand::thread_rng().fill(&mut random_buffer[..]);
+            random_buffer
+        };
+
+        volume
+            .write(
+                Block::new(0, BLOCK_SIZE.trailing_zeros()),
+                Bytes::from(random_buffer.clone()),
+            )?
+            .block_wait()?;
+
+        volume.deactivate()?.block_wait()?;
+
+        let mut active_check_counter = 0;
+        while volume.query_is_active()? {
```

Comment:
Did you see this loop activate.  I believe that each upstairs deactivate, if waited on, will not return until the deactivation has completed.  Since I believe that is what you do in the `volume.deactivate`, I would not think it would return from `volume.deactivate()?.block_wait()?;` until the deactivation is complete.  Maybe I'm not understanding how block_wait works?

Reply:
You're not wrong, this code is - this was put in previous to 0e27718de864ee59b6c1f17b65e0265737818424, so I ripped it out in aab698d